webrender/src/svg_renderer.rs

```diff
@@ -0,0 +1,144 @@
+                    }
+                    GeometryRequestMsg::Request(key, dimensions) => {
+                        let geometry = svg_geometries.get(&key)
+                            .expect("Requested geometries wasn't added!");
+                        let image_data = paint_svg(&context, &vg, geometry, dimensions);
+                        let data = ImageData::Raw(Arc::new(image_data));
+                        let descriptor = ImageDescriptor {
+                            format: ImageFormat::BGRA8,
+                            width: dimensions.width,
+                            height: dimensions.height,
+                            stride: None,
+                            offset: 0,
+                            is_opaque: false,
+                        };
+                        let msg = GeometryResultMsg::Image(key, data, descriptor);
+                        result_tx.send(msg).unwrap();
+                    }
+                    GeometryRequestMsg::EndFrame => {
+                        result_tx.send(GeometryResultMsg::EndFrame).unwrap();
+                    }
+                }
+            }
+        })
+        .expect("Thread spawning failed");
+    (msg_tx, result_rx)
+}
+
+fn paint_svg(context: &GLContext<NativeGLContext>,
+             vg: &Context,
+             geometry: &[GeometryItem],
+             dimensions: DeviceUintSize)
+             -> Vec<u8> {
+    context.gl().clear_color(1.0, 1.0, 1.0, 1.0);
+    context.gl().clear(gl::COLOR_BUFFER_BIT);
+
+    vg.begin_frame(VIEWPORT_WIDTH as u32, VIEWPORT_HEIGHT as u32, 1.0);
+    // the opengl y-coordinate is inverse with svg, so apply a transform here first.
+    let transform = Transform::new(1.0, 0., 0., -1.0, 0., VIEWPORT_HEIGHT as f32);
+    vg.transform(transform);
+    for item in geometry {
+        match item {
+            &GeometryItem::Shape(ref shape) => {
+                vg.begin_path();
+                for command in &shape.path {
+                    match command {
+                        &Command::MoveTo(point) => vg.move_to(point.x, point.y),
+                        &Command::LineTo(point) => vg.line_to(point.x, point.y),
+                        &Command::Arc(center, radius, start, end) => {
+                            let direction = if start < end {
+                                Winding::CCW
+                            } else {
+                                Winding::CW
+                            };
+                            vg.arc(center.x, center.y, radius, start, end, direction);
+                        }
+                    }
+                }
+                vg.close_path();
+                let color = Color::rgba(shape.fill.r, shape.fill.g, shape.fill.b, shape.fill.a);
+                vg.path_winding(Solidity::HOLE);
+                vg.fill_color(color);
+                vg.fill();
+            }
+        }
+    }
+    vg.end_frame();
+    context.gl().read_pixels(0,
```

Comment:
Well, I am not familiar with OpenGL and it took me sometime to work this out so I choose the easiest approach. Do you have suggestion on improve this? Or maybe we can do this to later, consider for now when the request dimension changes I simply delete previous texture from the cache and insert the new one, so it may be benefiting to use the ExternalImage api or wait #1367 work out. (I guess the won't be slower than skia? I don't know)